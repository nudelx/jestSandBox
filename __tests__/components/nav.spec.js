import React from "react"
import Enzyme, { shallow } from "enzyme"
import Adapter from "enzyme-adapter-react-16"
import Nav from "../../src/components/navigator"
import toJson from "enzyme-to-json"
Enzyme.configure({ adapter: new Adapter() })
const onClick = jest.fn()
let component = null

describe("Navigator component ", () => {
  beforeEach(() => {
    component = shallow(<Nav onClick={onClick} />)
  })

  it("should render without crashing", () => {
    expect(toJson(component)).toMatchSnapshot()
  })

  it("should have buttons with click function", () => {
    expect(toJson(component.find("li"))).toMatchSnapshot()
  })

  it("should pass the params to the click function", () => {
    component.find("li").map(item => item.simulate("click"))
    expect(onClick.mock.calls).toMatchSnapshot()
  })
})
