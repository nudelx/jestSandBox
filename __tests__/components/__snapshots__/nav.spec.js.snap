// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigator component  should have buttons with click function 1`] = `
Array [
  <li
    onClick={[Function]}
  >
    Page 1
  </li>,
  <li
    onClick={[Function]}
  >
    Page 2
  </li>,
  <li
    onClick={[Function]}
  >
    Page 3
  </li>,
]
`;

exports[`Navigator component  should pass the params to the click function 1`] = `
Array [
  Array [
    "page_1",
  ],
  Array [
    "page_2",
  ],
  Array [
    "page_3",
  ],
]
`;

exports[`Navigator component  should render without crashing 1`] = `
<div
  className="nav-bar"
>
  <ul>
    <li
      onClick={[Function]}
    >
      Page 1
    </li>
    <li
      onClick={[Function]}
    >
      Page 2
    </li>
    <li
      onClick={[Function]}
    >
      Page 3
    </li>
  </ul>
</div>
`;
